app/models/access.rb

```diff
@@ -0,0 +1,38 @@
+class Access < ApplicationRecord
+  ALLOWED_RESOURCE_TYPES = %w[Organization FacilityGroup Facility].freeze
+
+  belongs_to :user
+  belongs_to :resource, polymorphic: true, optional: true
+
+  enum role: {
+    super_admin: "super_admin",
+    admin: "admin",
+    analyst: "analyst"
+  }
+
+  validates :role, presence: true
+  validates :resource_type, inclusion: {in: ALLOWED_RESOURCE_TYPES}, allow_nil: true
+
+  class << self
```

Comment:
These methods are the workhorse for all scoping/authorization related work. It's basically just a walk-up-tree type query on the individual levels.